### DISCUSSION TITLE
Is it possible to use `Repo.iter_commits` or other API to get renames of files?

### DISCUSSION THREAD

**Discussion by Guo-Zhang** [gitpython-developers/GitPython#1477] [Q&A] [answered]
I want to track a file with its rename ones. For example, a file is called "test_lecture", and then rename to "python_data_analytics". I want to track all the commits for the files if possible.

I use `Repo.iter_commits` currently, which requires on `git rev-list` command, a complicated command(https://stackoverflow.com/questions/64397278/understanding-git-rev-list).

I have noticed that `git log` with `--follow` and `--find-renames`(https://www.zhihu.com/question/531483642/answer/2472733253) may work. Or are there any API based on it can be used?

**Comment by Byron** [comment, marked as answer]
I don't think that out of the box this is easy in `GitPython`, but it should be possible to use `git` directly and parse its output instead.
`GitPython` can make calling into `git` [more pythonic](https://gitpython.readthedocs.io/en/stable/tutorial.html#using-git-directly), too, but it won't change the fact that you have to deal with a steam of bytes to parse.
